src/views/DrawerView.tsx

```diff
@@ -246,67 +180,68 @@ export default class DrawerView extends React.PureComponent<Props, State> {
     }
   };
 
-  _setDrawerGestureRef = (ref: any) => {
+  private setDrawerGestureRef = (ref: PanGestureHandler | null) => {
     // @ts-ignore
     this.drawerGestureRef.current = ref;
   };
 
   render() {
-    const { navigation, screenProps } = this.props;
+    const { navigation } = this.props;
+    const {
+      drawerType,
+      drawerBackgroundColor,
+      overlayColor,
+      contentContainerStyle,
+      edgeWidth,
+      minSwipeDistance,
+      hideStatusBar,
+      statusBarAnimation,
+    } = this.props.navigationConfig;
     const activeKey = navigation.state.routes[navigation.state.index].key;
     const { drawerLockMode } = this.props.descriptors[activeKey].options;
 
+    const isOpen =
+      drawerLockMode === 'locked-closed'
+        ? false
+        : drawerLockMode === 'locked-open'
+        ? true
+        : this.props.navigation.state.isDrawerOpen;
+
     return (
-      <DrawerLayout
-        ref={(c: any) => {
-          this._drawer = c;
-        }}
-        onGestureRef={this._setDrawerGestureRef}
-        drawerLockMode={
-          drawerLockMode ||
-          (typeof screenProps === 'object' &&
-            screenProps != null &&
-            // @ts-ignore
-            screenProps.drawerLockMode) ||
-          this.props.navigationConfig.drawerLockMode
-        }
-        drawerBackgroundColor={
-          this.props.navigationConfig.drawerBackgroundColor
-        }
-        keyboardDismissMode={this.props.navigationConfig.keyboardDismissMode}
-        drawerWidth={this.state.drawerWidth}
-        onDrawerOpen={this._handleDrawerOpen}
-        onDrawerClose={this._handleDrawerClose}
-        onDrawerStateChanged={this._handleDrawerStateChange}
-        useNativeAnimations={this.props.navigationConfig.useNativeAnimations}
-        renderNavigationView={this._renderNavigationView}
-        drawerPosition={
-          this.props.navigationConfig.drawerPosition === 'right'
-            ? DrawerLayout.positions.Right
-            : DrawerLayout.positions.Left
-        }
-        /* props specific to react-native-gesture-handler/DrawerLayout */
-        drawerType={this.props.navigationConfig.drawerType}
-        edgeWidth={this.props.navigationConfig.edgeWidth}
-        hideStatusBar={this.props.navigationConfig.hideStatusBar}
-        statusBarAnimation={this.props.navigationConfig.statusBarAnimation}
-        minSwipeDistance={this.props.navigationConfig.minSwipeDistance}
-        overlayColor={this.props.navigationConfig.overlayColor}
-        drawerContainerStyle={this.props.navigationConfig.drawerContainerStyle}
-        contentContainerStyle={
-          this.props.navigationConfig.contentContainerStyle
-        }
-      >
-        <DrawerGestureContext.Provider value={this.drawerGestureRef}>
-          {this._renderContent()}
-        </DrawerGestureContext.Provider>
-      </DrawerLayout>
+      <DrawerGestureContext.Provider value={this.drawerGestureRef}>
+        <Drawer
+          open={isOpen}
```

Comment:
👌i love that this is a controlled prop now, it makes the implementation so much cleaner (like removing all of the genId stuff for actions) now that we can own the open state entirely in the router